packager/packager.cc

```diff
@@ -1576,14 +1669,34 @@ void validateLayering(const core::Context &ctx, const Import &i) {
                            thisPkg.show(ctx), "layer");
             e.addErrorLine(core::Loc(otherPkg.loc.file(), otherPkg.layer.value().second), "`{}`'s `{}` declared here",
                            otherPkg.show(ctx), "layer");
+            // TODO: Autocorrect to delete this import?
         }
     }
 
-    if (otherPkg.strictDependenciesLevel.value().first == core::packages::StrictDependenciesLevel::False) {
-        if (auto e = ctx.beginError(i.name.loc, core::errors::Packager::LayeringViolation)) {
-            e.setHeader("All of `{}`'s `{}`s must be `{}` or higher", thisPkg.show(ctx), "import", "layered");
+    core::packages::StrictDependenciesLevel otherPkgExpectedLevel = core::packages::StrictDependenciesLevel::Layered;
+
+    if (thisPkg.strictDependenciesLevel.value().first == core::packages::StrictDependenciesLevel::Dag) {
+        otherPkgExpectedLevel = core::packages::StrictDependenciesLevel::Dag;
+    }
+
+    if (otherPkg.strictDependenciesLevel.value().first < otherPkgExpectedLevel) {
+        if (auto e = ctx.beginError(i.name.loc, core::errors::Packager::StrictDependenciesViolation)) {
+            e.setHeader("All of `{}`'s `{}`s must be `{}` or higher", thisPkg.show(ctx), "import",
+                        strictDependenciesLevelToString(otherPkgExpectedLevel));
             e.addErrorLine(core::Loc(otherPkg.loc.file(), otherPkg.strictDependenciesLevel.value().second),
                            "`{}`'s `{}` level declared here", otherPkg.show(ctx), "strict_dependencies");
+            // TODO: Autocorrect to delete this import?
+        }
+    }
+
+    if (thisPkg.strictDependenciesLevel.value().first >= core::packages::StrictDependenciesLevel::LayeredDag) {
+        if (thisPkg.sccID == otherPkg.sccID) {
+            if (auto e = ctx.beginError(i.name.loc, core::errors::Packager::StrictDependenciesViolation)) {
+                e.setHeader("importing `{}` will put `{}` into a cycle, which is not valid at `{}` level `{}`",
```

Comment:
```suggestion
                e.setHeader("Importing `{}` will put `{}` into a cycle, which is not valid at `{}` level `{}`",
```

Reply:
I'm not sure I understand what you're getting at here. This error only occurs when the current package is in a cyle, ie. not when a dependency is in a cycle. So the user needs to fix all the places this specific error reports a cycle.

Is your point that they user will see this error, and then assume that all cycles are problematic, including ones where we don't report any errors?

Reply:
On second read I think the phrasing is good as is.